app/src/main/res/layout/sheet_custom_drug_entry.xml

```diff
@@ -35,80 +35,76 @@
     android:layout_gravity="center_vertical"
     android:layout_marginTop="@dimen/spacing_18"
     android:layout_marginEnd="@dimen/spacing_8"
-    android:paddingVertical="@dimen/spacing_10"
-    android:paddingHorizontal="@dimen/spacing_18"
     android:minWidth="@dimen/spacing_0"
     android:minHeight="@dimen/spacing_0"
+    android:paddingHorizontal="@dimen/spacing_18"
+    android:paddingVertical="@dimen/spacing_10"
     android:text="@string/custom_drug_entry_sheet_remove"
     app:layout_constraintEnd_toEndOf="parent"
     app:layout_constraintTop_toTopOf="parent"
     tools:visibility="visible" />
 
-  <LinearLayout
-    android:id="@+id/textFieldsLinearLayout"
-    android:layout_width="match_parent"
+  <com.google.android.material.textfield.TextInputLayout
+    android:id="@+id/dosageTextInputLayout"
+    android:layout_width="0dp"
     android:layout_height="wrap_content"
-    android:layout_marginHorizontal="@dimen/spacing_16"
     android:layout_marginStart="@dimen/spacing_24"
     android:layout_marginTop="@dimen/spacing_16"
-    android:layout_marginEnd="@dimen/spacing_24"
     android:layout_marginBottom="@dimen/spacing_16"
-    android:baselineAligned="true"
-    android:orientation="horizontal"
-    app:layout_constraintEnd_toEndOf="parent"
+    app:hintTextAppearance="?attr/textAppearanceBody2"
+    app:layout_constraintEnd_toStartOf="@+id/frequencyTextInputLayout"
+    app:layout_constraintHorizontal_weight="1"
+    app:layout_constraintLeft_toLeftOf="parent"
     app:layout_constraintStart_toStartOf="parent"
     app:layout_constraintTop_toBottomOf="@+id/titleTextView">
 
-    <com.google.android.material.textfield.TextInputLayout
-      android:layout_width="@dimen/spacing_0"
+    <com.google.android.material.textfield.TextInputEditText
+      android:id="@+id/drugDosageEditText"
+      android:layout_width="match_parent"
       android:layout_height="wrap_content"
-      android:layout_weight="1"
-      app:hintTextAppearance="?attr/textAppearanceBody2">
+      android:hint="@string/custom_drug_entry_sheet_dosage"
+      android:imeOptions="actionNext"
+      android:inputType="textCapWords"
+      android:textColor="@color/color_on_surface_67" />
+  </com.google.android.material.textfield.TextInputLayout>
 
-      <com.google.android.material.textfield.TextInputEditText
-        android:id="@+id/drugDosageEditText"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:hint="@string/custom_drug_entry_sheet_dosage"
-        android:imeOptions="actionNext"
-        android:inputType="textCapWords"
-        android:textColor="@color/color_on_surface_67" />
-
-    </com.google.android.material.textfield.TextInputLayout>
+  <com.google.android.material.textfield.TextInputLayout
```

Comment:
The frequency label color always seems to be the primary color. I have confirmed with design, and only text fields that are focused/active need to have label color as primary.

Also, the text color in designs has been updated PTAL.